src/concepts.hpp

```diff
@@ -120,6 +120,22 @@ concept Point =
   Subtractible<const T&, const T&, U> &&
   Subtractible<const T&, const U&, T>;
 
+template <Arithmetic, std::size_t>
+struct point;
+
+template <Arithmetic>
+struct dual;
+
+// TODO replace total ordering requirement with partial ordering?
+// TODO require Point to be rebindable? Otherwise we rebind into `opt::point`
```

Comment:
Yes, we need to be able to always create a dual from a Point in the "natural" way (if I understand correctly)